packages/core/src/custom-values.ts

```diff
@@ -154,6 +154,13 @@ export const CustomValueStrategy = {
                     } else {
                         resolvedValue = unbox(getStringValue(valueNode), !boxPrimitive);
                     }
+                } else if (typeof resolvedValue === 'string') {
+                    const parsedArg = postcssValueParser(resolvedValue).nodes[0];
```

Comment:
We want to ignore comments so the `nodes[0]` should be changed to take the first node that is not comment. I assume space cannot be the first token.

Reply:
the resolved values are already without any comments

Reply:
Currently referenced vars are stripped of comments, while the behavior of the top level is strange - it strip any comments before the referenced `value()` and keeps any after... while not related directly to this fix, I added another fix to strip any leftover comments.